### DISCUSSION TITLE
Github Action for MicroPython Code Check?

### DISCUSSION THREAD

**Discussion by matthias-bs** [micropython/micropython#10650] [General]
Hi,

is there an example Github Action CI workflow for MicroPython? I would like to use this as a code checker.

Thanks in advance!

Matthias

**Comment by dlech** [comment]
https://github.com/micropython/micropython/tree/master/.github/workflows

**Comment by matthias-bs** [comment]
Hmm, I saw this directory, but as far as I understand, it is for CI of the MicroPython environment/toolchain. What I am looking for is a workflow for MicroPython libraries/applications.

**Reply by jimmo** [reply]
@matthias-bs I'm not quite sure what you're asking? What do you mean by "code checker". If you mean code formatting (e.g. `black` in our case), then what @dlech linked to is what you want (specifically the `code_formatting.yml` workflow). We also do the same thing in micropython-lib: https://github.com/micropython/micropython-lib/tree/master/.github/workflows

**Comment by matthias-bs** [comment]
I would like to do syntax and coding-style checks.

**Reply by jimmo** [reply]
OK then the workflow linked above is exactly that.

**Reply by matthias-bs** [reply]
Thanks a lot!

**Comment by Josverl** [comment]
Furthermore, You can do more elaborate static code analysis using tools such as pylance, and codeql , when you add the applicable typestubs for your port/board of micropython

Apart from installing the micropython stubs, that is identical to running static code analysis on other python code

**Reply by matthias-bs** [reply]
That sounds good! How can I do this? Could you please provide any links/examples?
I've used pylint, but without micropython stubs.

**Reply by Josverl** [reply]
https://micropython-stubs.readthedocs.io/en/main/
